plugins/inputs/qbittorrent/qbittorrentdto.go

```diff
@@ -0,0 +1,530 @@
+		"priority":                    t.Priority,
+		"progress":                    t.Progress,
+		"ratio":                       t.Ratio,
+		"ratio_limit":                 t.RatioLimit,
+		"seeding_time":                t.SeedingTime,
+		"seeding_time_limit":          t.SeedingTimeLimit,
+		"seen_complete":               t.SeenComplete,
+		"size":                        t.Size,
+		"time_active":                 t.TimeActive,
+		"total_size":                  t.TotalSize,
+		"trackers_count":              t.TrackersCount,
+		"up_limit":                    t.UPLimit,
+		"uploaded":                    t.Uploaded,
+		"uploaded_session":            t.UploadedSession,
+		"upspeed":                     t.UPSpeed,
+	}
+}
+func (t Torrent) toTagsMap() map[string]string {
+	result := map[string]string{
+		"category":      t.Category,
+		"content_path":  t.ContentPath,
+		"download_path": t.DownloadPath,
+		"infohash_v1":   t.InfohashV1,
+		"infohash_v2":   t.InfohashV2,
+		"magnet_uri":    t.MagnetURI,
+		"name":          t.Name,
+		"save_path":     t.SavePath,
+		"state":         t.State,
+		"tags":          t.Tags,
+		"tracker":       t.Tracker,
+	}
+	if t.AutoTMM != nil {
+		result["auto_tmm"] = strconv.FormatBool(*t.AutoTMM)
+	}
+	if t.FLPiecePrio != nil {
+		result["fl_piece_prio"] = strconv.FormatBool(*t.FLPiecePrio)
+	}
+	if t.ForceStart != nil {
+		result["force_start"] = strconv.FormatBool(*t.ForceStart)
+	}
+	if t.SeqDownload != nil {
+		result["seq_download"] = strconv.FormatBool(*t.SeqDownload)
+	}
+	if t.SuperSeeding != nil {
+		result["super_seeding"] = strconv.FormatBool(*t.SuperSeeding)
+	}
+	return result
+}
+
+type MainData struct {
+	Categories        map[string]Category `json:"categories"`
+	CategoriesRemoved []string            `json:"categories_removed"`
+	FullUpdate        *bool               `json:"full_update"`
+	RID               int16               `json:"rid"`
+	ServerState       ServerState         `json:"server_state"`
+	Tags              []string            `json:"tags"`
+	TagsRemoved       []string            `json:"tags_removed"`
+	Torrents          map[string]Torrent  `json:"torrents"`
+	TorrentsRemoved   []string            `json:"torrents_removed"`
+	Trackers          map[string][]string `json:"trackers"`
+}
+
+func (m *MainData) partialUpdate(update *MainData) {
+	m.RID = update.RID
+	for k, v := range update.Categories {
+		category, exists := m.Categories[k]
+		if exists {
+			localV := v
+			(&category).partialUpdate(&localV)
+			m.Categories[k] = category
+		} else {
+			m.Categories[k] = v
+		}
+	}
+	for _, v := range update.CategoriesRemoved {
+		delete(m.Categories, v)
+	}
+	m.ServerState.partialUpdate(&update.ServerState)
+
+	m.Tags = append(m.Tags, update.Tags...)
+
+	if len(m.TagsRemoved) > 0 {
+		var removeTagsResult []string
+		for _, tag := range m.Tags {
+			found := false
+			for _, removedTag := range m.TagsRemoved {
+				if tag == removedTag {
+					found = true
+					break
+				}
+			}
+
+			if !found {
```

Comment:
It probably would be easier (and faster) to use a [filter](https://github.com/influxdata/telegraf/blob/master/filter/filter.go)...